### PR TITLE
Enhanced logger

### PR DESCRIPTION
**Pull request by aionJoey**
## Description

Added additional functionality and configuration to existing logger system;

- Allows user to enable logger in config.xml file
- Automatically rollover log files based on size (100MB) and time (daily)
- Groups archived log files into month folders
- Allows user to define log folder path
- Checks for invalid path name input (+ outputs error on terminal)

Fixes Issue # .

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] Bug fix.
- [x] New feature.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

- Tested file path and features manually
- Added test cases for file path output and configurations

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [x] New and existing tests pass locally with my changes.
- [x] I have added tests for my fix or feature.
- [ ] I have made appropriate changes to the corresponding documentation.
- [x] My code generates no new warnings.
- [x] Any dependent changes have been made.

### CODE REVIEW

File: modMcf/src/org/aion/mcf/config/CfgLog.java [hunk 1]
```diff
@@ -85,6 +99,21 @@ public String toXML() {
...
+            /** XML - Displays tag/entry in the config.xml */
+            xmlWriter.writeCharacters("\t\t");
+            xmlWriter.writeStartElement("log-file");
+            xmlWriter.writeCharacters(this.logFile + "");
+            xmlWriter.writeEndElement();
+            xmlWriter.writeCharacters("\r\n");
+
+            /** XML - Displays log-path in the config.xml */
+            xmlWriter.writeCharacters("\t\t");
+            xmlWriter.writeStartElement("log-path");
```
Comment (AlexandraRoatis) [inline modMcf/src/org/aion/mcf/config/CfgLog.java]: could you add comments here regarding the meaning of the configuration (similar to the ones in the `api` and `db`)

File: modBoot/src/org/aion/Aion.java [hunk 2]
```diff
@@ -74,31 +74,41 @@ public static void main(String args[]) throws InterruptedException {
...
+        if (!cfg.getLog().getLogFile()) {
+            System.out.println("Logger disabled; to enable please check log settings in config.xml\n");
+        } else if (!cfg.getLog().isValidPath() && cfg.getLog().getLogFile()) {
+            System.out.println("File path is invalid; please check log setting in config.xml\n");
+            return;
+        } else if (cfg.getLog().isValidPath() && cfg.getLog().getLogFile()) {
+            System.out.println("Logger file path: '" + cfg.getLog().getLogPath() + "'\n");
+        }
+
+        // If commit this out, the config setting will be ignore. all log module been set to "INFO" Level
```
Comment (AlexandraRoatis) [inline modBoot/src/org/aion/Aion.java]: not clear what that comment is mean to say

File: modLogger/src/org/aion/log/AionLoggerFactory.java [hunk 3]
```diff
@@ -70,12 +75,67 @@
...
+
+    public static void init(final Map<String, String> _logModules, boolean _logToFile, String _logToPath) {
 
         logModules = _logModules;
+        logFile = _logToFile;
+        logPath = _logToPath;
 
         loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
 
+        /** Toggles file appending configurations */
```
Comment (AlexandraRoatis) [inline modLogger/src/org/aion/log/AionLoggerFactory.java]: could you switch all the comments that use the java doc notation `/**` to the comment notation `/*` throughout all the files of the PR

File: modMcf/src/org/aion/mcf/config/CfgLog.java (hunk 1 above)
Comment (aionJoey) [reply modMcf/src/org/aion/mcf/config/CfgLog.java]: Added comments into the config.xml file

File: modBoot/src/org/aion/Aion.java (hunk 2 above)
Comment (aionJoey) [reply modBoot/src/org/aion/Aion.java]: Removed comment

File: modLogger/src/org/aion/log/AionLoggerFactory.java (hunk 3 above)
Comment (aionJoey) [reply modLogger/src/org/aion/log/AionLoggerFactory.java]: Changed the comment notation and added the license headers to relevant files

File: modMcf/src/org/aion/mcf/config/CfgLog.java [hunk 4]
```diff
@@ -104,6 +104,8 @@ public String toXML() {
              * Boolean value to allow logger to be toggled ON and OFF
              */
             xmlWriter.writeCharacters("\t\t");
+            xmlWriter.writeCharacters("<!--Enable/Disable logback service; if disabled, output will not be logged -->");
```
Comment (AlexandraRoatis) [inline modMcf/src/org/aion/mcf/config/CfgLog.java]: you need to use `writeComment` here like in https://github.com/aionnetwork/aion/blob/1bbeec193eda2a915233faa51aa69680d5bcfd43/modMcf/src/org/aion/mcf/config/CfgDb.java#L195-L200

File: modMcf/src/org/aion/mcf/config/CfgLog.java [hunk 5]
```diff
@@ -114,6 +116,8 @@ public String toXML() {
              * String value to determine the folder path for log files
              */
             xmlWriter.writeCharacters("\t\t");
+            xmlWriter.writeCharacters("<!--Sets the physical location on disk where log files will be stored.-->");
```
Comment (AlexandraRoatis) [inline modMcf/src/org/aion/mcf/config/CfgLog.java]: same as before, need to use `writeComment` and remove the comment tags

**Pull request merged**